common.sh

```diff
@@ -187,9 +187,40 @@ create_instance()
 get_instance_ip()
 {
     execution_seq=$((${execution_seq}+1))
-    INSTANCE_IPS=$(aws ec2 describe-instances --instance-ids ${INSTANCE_IDS[@]} \
-                        --query "Reservations[*].Instances[*].PrivateIpAddress" \
-                        --output=text)
+    local retry=20
+    local sleep_time=10
+    local ret=0
+    local bash_option=$-
+    local restore_e=0
+    local get_instance_ip_succeed=0
+    local instance_ips_array=()
+    if [[ $bash_option =~ e ]]; then
+        restore_e=1
+        set +e
+    fi
+    while [ $retry -ge 0 ]; do
+        INSTANCE_IPS=$(aws ec2 describe-instances --instance-ids ${INSTANCE_IDS[@]} \
+                            --query "Reservations[*].Instances[*].PrivateIpAddress" \
+                            --output=text)
+        ret=$?
+        instance_ips_array=($INSTANCE_IPS)
+        if [[ $ret -eq 0 && -n "$INSTANCE_IPS" && ${#instance_ips_array[@]} -eq ${#INSTANCE_IDS[@]} ]]; then
```

Comment:
you don't actually need the quotes here, but they also do no harm.  The quotes are normally needed because of shell escaping rules when launching a process (`[` or `test` must behave like subprocesses, because that's what they are in pure sh, but `[[` doesn't have to, because it's a bashism with no sh equivalent).

I'd have converted the whole subroutine to bash-style conditionals, but not critical.

Reply:
Thank you! I removed the quotes in this line, and converted all 3 subroutines in this PR to bash-style conditions.